php/setup.php

```diff
@@ -0,0 +1,218 @@
+        private readonly string $applicationBaseUrl,
+    ) {
+        $this->client = new FusionAuthClient($apiKey, $baseUrl);
+    }
+
+    public function run(): stdClass
+    {
+        echo 'Retrieving Tenants... ';
+        $tenant = $this->getTenant();
+        echo 'OK' . PHP_EOL;
+
+        echo 'Patching Tenant... ';
+        $this->patchTenant($tenant);
+        echo 'OK' . PHP_EOL;
+
+        echo 'Generating RSA Key... ';
+        $this->generateKey();
+        echo 'OK' . PHP_EOL;
+
+        echo 'Creating Application... ';
+        $this->createApplication();
+        echo 'OK' . PHP_EOL;
+
+        echo 'Retrieving Users... ';
+        $user = $this->getUser();
+        echo 'OK' . PHP_EOL;
+
+        echo "Patching User {$user->email}... ";
+        $this->patchUser($user);
+        echo 'OK' . PHP_EOL;
+
+        echo 'Registering User... ';
+        $this->registerUser($user);
+        echo 'OK' . PHP_EOL;
+
+        return $user;
+    }
+
+    private function handleResponse(string $method, ClientResponse $response): stdClass
+    {
+        if (!$response->wasSuccessful()) {
+            $message = 'An unknown error occurred. Please check the API Key and Base URL for your FusionAuth instance.';
+            $error = $response->errorResponse;
+            if (!empty($error)) {
+                $message = PHP_EOL . json_encode($error, JSON_PRETTY_PRINT);
+            }
+            throw new RuntimeException("Error while {$method}: {$message}", $response->status ?: 0);
+        }
+        return $response->successResponse ?? new stdClass();
+    }
+
+    private function getTenant(): stdClass
+    {
+        $response = $this->handleResponse(
+            'Retrieving Tenants',
+            $this->client->retrieveTenants()
+        );
+        if (empty($response->tenants)) {
+            throw new RuntimeException("Couldn't find any tenants");
+        }
+
+        return $response->tenants[0];
+    }
+
+    private function patchTenant(stdClass $tenant): void
+    {
+        $this->handleResponse(
+            'Patching Tenant',
+            $this->client->patchTenant(
+                $tenant->id,
+                ['tenant' => ['issuer' => $this->baseUrl]]
+            )
+        );
+    }
+
+    private function generateKey(): void
+    {
+        $this->handleResponse(
+            'Generating API Key',
+            $this->client->generateKey(
+                $this->rsaKeyId,
+                [
+                    'key' => [
+                        'algorithm' => 'RS256',
+                        'name'      => "For {$this->applicationName}",
+                        'length'    => 2048,
+                    ],
+                ]
+            )
+        );
+    }
+
+    private function createApplication(): void
+    {
+        $application = [
+            'name'               => $this->applicationName,
+            'oauthConfiguration' => [
+                'authorizedRedirectURLs' => [$this->applicationBaseUrl],
```

Comment:
Is this correct? We redirect to `http://localhost:9011`?

Reply:
No, `$this->applicationBaseUrl` is actually `http://localhost:8080` as defined on line 19.

`http://localhost:9011` is actually `$this->baseUrl', which is used to instantiate the `FusionAuthClient` and to set the issuer for the tenant.

Reply:
my bad.